agent/agentbootstrap/bootstrap.go

```diff
@@ -388,7 +388,7 @@ func getEnviron(
 		Cloud:          cloudSpec,
 		Config:         modelConfig,
 	}
-	if cloudSpec.Type == cloud.CloudTypeCAAS {
```

Comment:
Could we make this a method on `Type` instead?

Reply:
It is a good idea.
I tried to move this to cloud.Type. But I found I have to touch hundreds of files (from `cmd` to `api/apiserver` then `state`).
Given this PR is already not small. It might be better to do this change in a separate PR.